src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassaymetadata/CorrelationAssayMetadataPlugin.java

```diff
@@ -0,0 +1,151 @@
+
+
+    return List.of(new StreamSpec(INPUT_DATA, getConfig().getCollectionVariable1().getEntityId())
+        .addVars(getUtil().getChildrenVariables(collectionVariable1VarSpec))
+        .addVars(metadataVariables)
+      );
+  }
+
+  @Override
+  protected void execute() {
+
+    CorrelationComputeConfig computeConfig = getConfig();
+    PluginUtil util = getUtil();
+    ReferenceMetadata meta = getContext().getReferenceMetadata();
+
+    // Get compute parameters
+    String method = computeConfig.getCorrelationMethod().getValue();
+    CollectionSpec collectionVariable1 = computeConfig.getCollectionVariable1();
+    String entityId = collectionVariable1.getEntityId();
+
+    // Wrangle into helpful types
+    EntityDef entity = meta.getEntity(entityId).orElseThrow();
+    VariableDef computeEntityIdVarSpec = util.getEntityIdVarSpec(entityId);
+    String computeEntityIdColName = util.toColNameOrEmpty(computeEntityIdVarSpec);
+    VariableSpec collectionVariable1VarSpec = VariableDef.newVariableSpec(entityId, collectionVariable1.getCollectionId());
+
+    // Get record id columns
+    List<VariableDef> idColumns = new ArrayList<>();
+    for (EntityDef ancestor : meta.getAncestors(entity)) {
+      idColumns.add(ancestor.getIdColumnDef());
+    }
+
+    HashMap<String, InputStream> dataStream = new HashMap<>();
+    dataStream.put(INPUT_DATA, getWorkspace().openStream(INPUT_DATA));
+
+
+    // Get stream of all ancestors and their variables
+    Stream<VariableDef> descendantVariableStream = meta.getAncestors(entity).get(1).getVariables().stream();
+    List<VariableDef> metadataVariables = descendantVariableStream.filter(var -> var.getDataShape() == APIVariableDataShape.CONTINUOUS).filter(var -> !var.getVariableId().contains("_stable_id")).toList();
+
+
+    RServe.useRConnectionWithRemoteFiles(dataStream, connection -> {
+      connection.voidEval("print('starting correlation computation')");
+
+      // Read in the abundance data
+      List<VariableSpec> abundanceInputVars = ListBuilder.asList(computeEntityIdVarSpec);
+      abundanceInputVars.addAll(util.getChildrenVariables(collectionVariable1VarSpec));
+      abundanceInputVars.addAll(idColumns);
+      connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, abundanceInputVars));
+      connection.voidEval("abundanceData <- " + INPUT_DATA); // Renaming here so we can go get the sampleMetadata later
+
+      // Read in the sample metadata
+      List <VariableSpec> metadataInputVars = ListBuilder.asList(computeEntityIdVarSpec);
+      metadataInputVars.addAll(metadataVariables);
+      metadataInputVars.addAll(idColumns);
+      connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, metadataInputVars));
+      connection.voidEval("sampleMetadata <- " + INPUT_DATA); 
+
+      // Turn the list of id columns into an array of strings for R
```

Comment:
pretty sure we have a helper in edacommon for this now

Reply:
.... pretty sure i even put it there...